src/main/java/com/atypon/wayf/facade/impl/PublisherFacadeImpl.java

```diff
@@ -115,7 +116,22 @@ public PublisherFacadeImpl() {
 
     @Override
     public Observable<Publisher> filter(PublisherQuery filter) {
-        return publisherDao.filter(filter);
+
+        boolean isAdminView = PublisherQuery.ADMIN_VIEW.equals(filter.getView());
+
+        if (isAdminView) {
+            AuthenticatedEntity.authenticatedAsAdmin(RequestContextAccessor.get().getAuthenticated());
+        }
+
+        return publisherDao.filter(filter).flatMap(publisher -> {
+            if (!isAdminView) {
+                publisher.setWidgetLocation(null);
+            } else {
+                publisher.setToken((AuthorizationToken) authenticationFacade.getCredentialsForAuthenticatable(publisher)
```

Comment:
Looks good. For now we can leave this call as is but for long term performance we might want to batch the authenticatables (publishers) so that we can avoid multiple trips to the DB.